theme/src/components/code.js

```diff
@@ -6,62 +6,142 @@ import React from 'react'
 import ClipboardCopy from './clipboard-copy'
 import LiveCode from './live-code'
 
-function Code({className, children, live, noinline}) {
+function Code({className, children, live, noinline, metastring}) {
   const language = className ? className.replace(/language-/, '') : ''
   const code = children.trim()
+  const shouldHighlight = calculateLinesToHighlight(metastring)
+  const {title} = extractProperties(metastring)
```

Comment:
Not sure if this utility methods should go to another file instead of being in the component, whatever you normally do is fine